src/gui/models/dissolveModel.h

```diff
@@ -0,0 +1,104 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2023 Team Dissolve and contributors
+
+#pragma once
+
+#include "gui/models/atomTypeModel.h"
+#include "gui/models/configurationModel.h"
+#include "gui/models/masterTermTreeModel.h"
+#include "gui/models/speciesModel.h"
+#include "main/dissolve.h"
+#include "templates/optionalRef.h"
+#include <memory>
+
+class DissolveModel : public QObject
+{
+    Q_OBJECT
+    // The Atom Type Model
+    Q_PROPERTY(AtomTypeModel *atomTypesModel READ atomTypesModel NOTIFY atomTypesChanged)
+    // The number of atom types
+    Q_PROPERTY(int nAtomTypes READ nAtomTypes NOTIFY modelsUpdated)
+    // The Master Bond Model
+    Q_PROPERTY(const MasterBondModel *masterBondsModel READ masterBondsModel NOTIFY mastersChanged)
+    // The number of master bonds
+    Q_PROPERTY(int nMasterBonds READ nMasterBonds NOTIFY modelsUpdated)
+    // The Master Angle Model
+    Q_PROPERTY(const MasterAngleModel *masterAnglesModel READ masterAnglesModel NOTIFY mastersChanged)
+    // The number of master angles
+    Q_PROPERTY(int nMasterAngles READ nMasterAngles NOTIFY modelsUpdated)
+    // The Master Torsion Model
+    Q_PROPERTY(const MasterTorsionModel *masterTorsionsModel READ masterTorsionsModel NOTIFY mastersChanged)
+    // The number of master torsions
+    Q_PROPERTY(int nMasterTorsions READ nMasterTorsions NOTIFY modelsUpdated)
+    // The Master Improper Model
+    Q_PROPERTY(const MasterImproperModel *masterImpropersModel READ masterImpropersModel NOTIFY mastersChanged)
+    // The number of master impropers
+    Q_PROPERTY(int nMasterImpropers READ nMasterImpropers NOTIFY modelsUpdated)
```

Comment:
```suggestion
    Q_PROPERTY(AtomTypeModel *atomTypesModel READ atomTypesModel NOTIFY atomTypesChanged)
    // The Master Bond Model
    Q_PROPERTY(const MasterBondModel *masterBondsModel READ masterBondsModel NOTIFY mastersChanged)
    // The Master Angle Model
    Q_PROPERTY(const MasterAngleModel *masterAnglesModel READ masterAnglesModel NOTIFY mastersChanged)
    // The Master Torsion Model
    Q_PROPERTY(const MasterTorsionModel *masterTorsionsModel READ masterTorsionsModel NOTIFY mastersChanged)
    // The Master Improper Model
    Q_PROPERTY(const MasterImproperModel *masterImpropersModel READ masterImpropersModel NOTIFY mastersChanged)
```

I get into a discussion lower down on why we should get rid of the count properties.